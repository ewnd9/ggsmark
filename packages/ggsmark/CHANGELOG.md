# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.10.3](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.10.2...ggsmark@0.10.3) (2020-09-20)

**Note:** Version bump only for package ggsmark





## [0.10.2](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.10.1...ggsmark@0.10.2) (2020-09-19)

**Note:** Version bump only for package ggsmark





## [0.10.1](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.10.0...ggsmark@0.10.1) (2020-09-19)


### Bug Fixes

* YouTube embed short-hand link ([6429f27](https://github.com/johnnyhuy/ggsmark/commit/6429f27065401beabbba9a549757933d0a113abb))





# [0.10.0](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.9.0...ggsmark@0.10.0) (2020-09-19)


### Bug Fixes

* removed unused package ([52a7562](https://github.com/johnnyhuy/ggsmark/commit/52a7562ac6cf671690d1c432cb9ff0ac7ea9e9c9))
* Ruby fence test output ([a47eb75](https://github.com/johnnyhuy/ggsmark/commit/a47eb75a2e772fa3b6b2070f914931907c7774eb))


### Features

* added code highlighting ([9c836a5](https://github.com/johnnyhuy/ggsmark/commit/9c836a52f9fbbb2849ea2c27f2755e80d0787e0f))





# [0.9.0](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.8.11...ggsmark@0.9.0) (2020-09-15)


### Bug Fixes

* get iframes working with Twitch clips ([b466af3](https://github.com/johnnyhuy/ggsmark/commit/b466af387300dcc00e2975c23a4623756a89bc68))
* snapshot bug ([90f479b](https://github.com/johnnyhuy/ggsmark/commit/90f479b8e10b6824fa99af27339e9f512d5af29a))


### Features

* encode parents into a URL string ([cf2f3bf](https://github.com/johnnyhuy/ggsmark/commit/cf2f3bfb88836de7c1ce71ce3704a4c46320d717))
* expose options ([259f1b5](https://github.com/johnnyhuy/ggsmark/commit/259f1b525f2d9a379a7bec8a6dc529f424d59de9))
* show clips, channels and other stuff ([b2c84e0](https://github.com/johnnyhuy/ggsmark/commit/b2c84e09372fe24daf377f6b4915f84ff511e48e))





## [0.8.11](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.8.10...ggsmark@0.8.11) (2020-09-14)

**Note:** Version bump only for package ggsmark





## [0.8.10](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.8.9...ggsmark@0.8.10) (2020-09-14)


### Bug Fixes

* added browser to dist ([fe7a4db](https://github.com/johnnyhuy/ggsmark/commit/fe7a4db9a9be13ea4d7d93b24215cd89cde13bad))





## [0.8.9](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.8.8...ggsmark@0.8.9) (2020-09-13)


### Bug Fixes

* Added runtime ([46671d9](https://github.com/johnnyhuy/ggsmark/commit/46671d96a8687a6c951b20b0e49cc6f6971bb74b))
* added the regenerator package properly ([17a5dcf](https://github.com/johnnyhuy/ggsmark/commit/17a5dcfd4fd163bf53f3233c5460af6befe4d099))





## [0.8.8](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.8.7...ggsmark@0.8.8) (2020-09-13)


### Bug Fixes

* try dist again! ([f435fc6](https://github.com/johnnyhuy/ggsmark/commit/f435fc65d3fa5698eb70f8be3b9a91454bd50efc))





## [0.8.7](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.8.6...ggsmark@0.8.7) (2020-09-13)


### Bug Fixes

* move index back to src folder ([b19ac59](https://github.com/johnnyhuy/ggsmark/commit/b19ac590f62a670e122a4af6b42161a8d168ad6d))
* src path ([820e5b9](https://github.com/johnnyhuy/ggsmark/commit/820e5b9c108a03f2d9d72e6dbd4989c81a601bd6))





## [0.8.6](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.8.5...ggsmark@0.8.6) (2020-09-13)

**Note:** Version bump only for package ggsmark





## [0.8.5](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.8.4...ggsmark@0.8.5) (2020-09-13)


### Bug Fixes

* remove core-js ([cdead31](https://github.com/johnnyhuy/ggsmark/commit/cdead31ce85d614ba4b9f147e11cb5472af6c362))





## [0.8.4](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.8.3...ggsmark@0.8.4) (2020-09-13)


### Bug Fixes

* try dist in browser ([aacd213](https://github.com/johnnyhuy/ggsmark/commit/aacd21371db48e1d094a4d64127af0703a919f4e))





## [0.8.3](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.8.2...ggsmark@0.8.3) (2020-09-13)


### Bug Fixes

* move index out to main folder ([ee60135](https://github.com/johnnyhuy/ggsmark/commit/ee60135b993d26a7c139046ea5f9c36f1e5f8b9c))





## [0.8.2](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.8.1...ggsmark@0.8.2) (2020-09-13)


### Bug Fixes

* Revert "fix: try dist" ([8d6d36e](https://github.com/johnnyhuy/ggsmark/commit/8d6d36e60408532a9218b5db0597540b8bf33bdb))
* Revert "fix: Use dist" ([f838486](https://github.com/johnnyhuy/ggsmark/commit/f8384861d7b043439a67dc903a9780dfafa8267e))
* try dist ([d02b34c](https://github.com/johnnyhuy/ggsmark/commit/d02b34c43b9d02ae0ee156113426954e1d0b5bc3))
* try src folder again ([5329336](https://github.com/johnnyhuy/ggsmark/commit/5329336cbb7456bc9fd844fcc43bbaa6c1fcc169))
* Use dist ([e030308](https://github.com/johnnyhuy/ggsmark/commit/e03030845ada2600a25b0bd894844eedb4fc1358))





## [0.8.1](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.8.0...ggsmark@0.8.1) (2020-09-13)

**Note:** Version bump only for package ggsmark





# [0.8.0](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.7.0...ggsmark@0.8.0) (2020-09-13)


### Features

* added core JS ([1d9e424](https://github.com/johnnyhuy/ggsmark/commit/1d9e424426cdf2ddae38c817e3d3f23d436cfca4))
* put packages in src folder ([d10a8bb](https://github.com/johnnyhuy/ggsmark/commit/d10a8bb233cf6140a0d0f37b2f8ae2670eeefe2a))





# [0.7.0](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.6.0...ggsmark@0.7.0) (2020-09-13)


### Features

* Added browser and module files ([9581389](https://github.com/johnnyhuy/ggsmark/commit/958138995ac17ed1eb3a9b13d2abb8b2e42521d5))





# [0.6.0](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.5.1...ggsmark@0.6.0) (2020-09-13)


### Bug Fixes

* don't ignore tests in main Babel ([bf2ecd2](https://github.com/johnnyhuy/ggsmark/commit/bf2ecd21c81d11522b13b3631bfd1ef877244271))


### Features

* Export to index JS ([08a5dc0](https://github.com/johnnyhuy/ggsmark/commit/08a5dc0e5f4278bf7fbccf23eb99647c5a82dbc3))





## [0.5.1](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.5.0...ggsmark@0.5.1) (2020-09-13)

**Note:** Version bump only for package ggsmark





# [0.5.0](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.4.1...ggsmark@0.5.0) (2020-09-13)


### Features

* added spoilers ([ec91cf1](https://github.com/johnnyhuy/ggsmark/commit/ec91cf11cc1f9e00eefd77310bb64d4fd28fefa5))





## [0.4.1](https://github.com/johnnyhuy/ggsmark/compare/ggsmark@0.4.0...ggsmark@0.4.1) (2020-09-13)

**Note:** Version bump only for package ggsmark
