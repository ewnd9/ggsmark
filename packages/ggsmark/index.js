import unified from 'unified'
import markdown from 'remark-parse'
import stringify from 'rehype-stringify'
import rehype from 'remark-rehype'
import iframe from 'remark-iframes'
import align from 'remark-text-alignment'

// Import this since remark-iframe needs it
import 'regenerator-runtime/runtime'

export default (text) => {
  return unified()
    .use(markdown, {
      blocks: []
    })
    .use(iframe, {
      'www.youtube.com': {
        tag: 'iframe',
        width: 560,
        height: 315,
        disabled: false,
        replace: [
          ['watch?v=', 'embed/'],
          ['http://', 'https://']
        ],
        thumbnail: {
          format: 'http://img.youtube.com/vi/{id}/0.jpg',
          id: '.+/(.+)$'
        },
        removeAfter: '&'
      },
      'youtu.be': {
        tag: 'iframe',
        width: 560,
        height: 315,
        disabled: false,
        replace: [
          ['watch?v=', 'embed/'],
          ['http://', 'https://']
        ],
        thumbnail: {
          format: 'http://img.youtube.com/vi/{id}/0.jpg',
          id: '.+/(.+)$'
        },
        removeAfter: '&'
      },
      'soundcloud.com': {
        tag: 'iframe',
        width: '100%',
        height: 150,
        disabled: false,
        replace: [
          [
            'soundcloud.com/',
            'w.soundcloud.com/player/?visual=true&url=https://soundcloud.com/'
          ],
          ['http://', 'https://']
        ]
      },
      'www.clips.twitch.tv': {
        tag: 'iframe',
        width: 620,
        height: 378,
        disabled: false,
        replace: [
          [
            'https://clips.twitch.tv/',
            'https://clips.twitch.tv/embed?parent=ggs.sx&clip='
          ],
          ['http://', 'https://']
        ]

      },
      'clips.twitch.tv': {
        tag: 'iframe',
        width: 620,
        height: 378,
        disabled: false,
        replace: [
          [
            'https://clips.twitch.tv',
            'https://clips.twitch.tv/embed?parent=ggs.sx&clip='
          ],
          ['http://', 'https://']
        ]
      },
      'medal.tv': {
        tag: 'iframe',
        width: 360,
        height: 640,
        disabled: false,
        replace: [
          ['http://', 'https://']
        ]
      },
    })
    .use(align, {
      left: 'align-left',
      center: 'align-center',
      right: 'align-right'
    })
    .use(rehype)
    .use(stringify)
    .processSync(text)
    .toString()
}
