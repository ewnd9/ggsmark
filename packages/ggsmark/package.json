{
  "name": "ggsmark",
  "version": "0.10.3",
  "description": "Markdown extension for the Gentlemen's Gaming Society website.",
  "repository": "https://github.com/johnnyhuy/ggsmark.git",
  "author": "Johnny Huynh <info@johnnyhuy.com>",
  "license": "MIT",
  "dependencies": {
    "deepmerge": "^4.2.2",
    "hast-util-sanitize": "^3.0.0",
    "regenerator-runtime": "^0.13.7",
    "rehype-highlight": "^4.0.0",
    "rehype-sanitize": "^4.0.0",
    "rehype-stringify": "^8.0.0",
    "remark-color-text": "^0.4.0",
    "remark-iframes": "^4.0.4",
    "remark-parse": "^8.0.3",
    "remark-rehype": "^7.0.0",
    "remark-spoilers": "^0.5.1",
    "remark-text-alignment": "^0.5.11",
    "unified": "^9.2.0"
  },
  "main": "dist/index.js",
  "browser": "dist/index.js",
  "module": "src/index.js",
  "scripts": {
    "build": "cross-env BABEL_ENV=production babel src/index.js --config-file ../../babel.config.js --ignore '**/*.spec.js,**/*.test.js' --out-dir dist"
  }
}
