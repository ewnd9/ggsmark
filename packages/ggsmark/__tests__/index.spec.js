import dedent from 'dedent'
import ggsmark from '../src'

describe('should have github-like markdown', () => {
  test('strikethrough', () => {
    // Arrange
    let string = dedent`
    ~~shit~~
    `
    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toBe(dedent`
    <p><del>shit</del></p>
    `)
  })

  test('auto-link', () => {
    // Arrange
    let string = dedent`
    visit https://ggs.sx/
    `
    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toBe(dedent`
    <p>visit <a href="https://ggs.sx/">https://ggs.sx/</a></p>
    `)
  })

  test('task lists', () => {
    // Arrange
    let string = dedent`
    - [x] Do work
    - [ ] Get a life
    - [ ] Open book
    `
    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toBe(dedent`
    <ul class="contains-task-list">
    <li class="task-list-item"><input type="checkbox" checked disabled> Do work</li>
    <li class="task-list-item"><input type="checkbox" disabled> Get a life</li>
    <li class="task-list-item"><input type="checkbox" disabled> Open book</li>
    </ul>
    `)
  })

  test('tables', () => {
    // Arrange
    let string = dedent`
    | Command | Description |
    | --- | --- |
    | git status | List all new or modified files |
    | git diff | Show file differences that haven't been staged |
    `
    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toBe(dedent`
    <table>
    <thead>
    <tr>
    <th>Command</th>
    <th>Description</th>
    </tr>
    </thead>
    <tbody>
    <tr>
    <td>git status</td>
    <td>List all new or modified files</td>
    </tr>
    <tr>
    <td>git diff</td>
    <td>Show file differences that haven't been staged</td>
    </tr>
    </tbody>
    </table>
    `)
  })

  test('code fencing', () => {
    // Arrange
    let string = dedent`
    \`\`\`ruby
    require 'redcarpet'
    markdown = Redcarpet.new("Hello World!")
    puts markdown.to_html
    \`\`\`
    `
    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toBe(dedent`
    <pre><code class="hljs language-ruby"><span class="hljs-keyword">require</span> <span class="hljs-string">'redcarpet'</span>
    markdown = Redcarpet.new(<span class="hljs-string">"Hello World!"</span>)
    puts markdown.to_html
    </code></pre>
    `)
  })
})

describe('render soundcloud blocks', () => {
  test('single line', () => {
    // Arrange
    let string = dedent`
    !(https://soundcloud.com/iamcardib/wap-feat-megan-thee-stallion)
    `

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('should not show iframe inline', () => {
    // Arrange
    let string =
      '**bold text** before text !(https://soundcloud.com/iamcardib/wap-feat-megan-thee-stallion) after text **bold**'

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('multiple occurrences', () => {
    // Arrange
    let string = dedent`
    !(https://soundcloud.com/iamcardib/wap-feat-megan-thee-stallion)
    !(https://soundcloud.com/iamcardib/wap-feat-megan-thee-stallion)
    !(https://soundcloud.com/iamcardib/wap-feat-megan-thee-stallion)
    `

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toMatchSnapshot()
  })
})

describe('render youtube blocks', () => {
  test('repeated youtube with text before and after', () => {
    // Arrange
    let string = dedent`
    **bold** string before comyoutube !(http://www.youtube.com/watch?v=52c_QSg64fs) after youtube !(http://www.youtube.com/watch?v=waefawefwaef) *italics*
    soft new line
    new line
    `

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('short-hand youtube', () => {
    // Arrange
    let string = dedent`
    !(http://www.youtu.be/52c_QSg64fs)
    `

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('repeated youtube', () => {
    // Arrange
    let string = dedent`
    !(http://www.youtube.com/watch?v=52c_QSg64fs)
    !(http://www.youtube.com/watch?v=52c_QSg64fs)
    !(http://www.youshit.com)
    `

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('youtube with timestamp', () => {
    // Arrange
    let string = dedent`!(https://youtu.be/IJhgZBn-LHg?t=712)`

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toMatchSnapshot()
  })
})

describe('render left alignment text', () => {
  test('single line', () => {
    // Arrange
    let string = dedent`
    <-LeftText<-
    `

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('multi line', () => {
    // Arrange
    let string = dedent`
    <-This is a multiple line texts\n 2nd line here<-
    `

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toMatchSnapshot()
  })
})

describe('render center alignment text', () => {
  test('single line', () => {
    // Arrange
    let string = dedent`
    ->CenterText<-
    `

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toMatchSnapshot()
  })
  test('multi line', () => {
    // Arrange
    let string = dedent`
    ->This is a multiple line texts\n 2nd line here<-
    `

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toMatchSnapshot()
  })
})

describe('render right alignment text', () => {
  test('single line', () => {
    // Arrange
    let string = dedent`
    ->RightText->
    `

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('multi line', () => {
    // Arrange
    let string = dedent`
    ->This is a multiple line texts\n 2nd line here->
    `

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toMatchSnapshot()
  })
})

describe('do not render custom html', () => {
  test('span', () => {
    // Arrange
    let string = dedent`
    <span style="color: red">Test</span>
    `
    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toBe(dedent`
    <p>Test</p>
    `)
  })

  test('heading', () => {
    // Arrange
    let string = dedent`
    <h3 style="color: red">Test</h3>
    `
    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toBe(dedent`
    <p>Test</p>
    `)
  })

  test('div', () => {
    // Arrange
    let string = dedent`
    <div style="color: red">Test</div>
    `
    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toBe(dedent`
    <p>Test</p>
    `)
  })
})

describe('render twitch video blocks', () => {
  test('single line', () => {
    // Arrange
    let string = dedent`
    !(https://clips.twitch.tv/LovelyAstuteCoffeeImGlitch)
    `

    // Act
    let result = ggsmark(string, { twitchParents: ['ggs.sx', 'example.com'] })

    // Assert
    expect(result).toBe(dedent`
    <iframe src="https://clips.twitch.tv/embed?parent=ggs.sx&#x26;parent=example.com&#x26;clip=LovelyAstuteCoffeeImGlitch" width="560" height="315" allowfullscreen frameborder="0"></iframe>
    `)
  })

  test('should not show iframe inline', () => {
    // Arrange
    let string = dedent`
    **bold text** before text !(https://clips.twitch.tv/LovelyAstuteCoffeeImGlitch) after text **bold**
    `

    // Act
    let result = ggsmark(string, { twitchParents: ['ggs.sx', 'example.com'] })

    // Assert
    expect(result).toBe(dedent`
    <p><strong>bold text</strong> before text !(<a href="https://clips.twitch.tv/LovelyAstuteCoffeeImGlitch">https://clips.twitch.tv/LovelyAstuteCoffeeImGlitch</a>) after text <strong>bold</strong></p>
    `)
  })

  test('multiple occurrences', () => {
    // Arrange
    let string = dedent`
    !(https://clips.twitch.tv/LovelyAstuteCoffeeImGlitch)
    !(https://clips.twitch.tv/LovelyAstuteCoffeeImGlitch)
    !(https://clips.twitch.tv/LovelyAstuteCoffeeImGlitch)
    `

    // Act
    let result = ggsmark(string, { twitchParents: ['ggs.sx', 'example.com'] })

    // Assert
    expect(result).toBe(dedent`
    <iframe src="https://clips.twitch.tv/embed?parent=ggs.sx&#x26;parent=example.com&#x26;clip=LovelyAstuteCoffeeImGlitch" width="560" height="315" allowfullscreen frameborder="0"></iframe>
    <iframe src="https://clips.twitch.tv/embed?parent=ggs.sx&#x26;parent=example.com&#x26;clip=LovelyAstuteCoffeeImGlitch" width="560" height="315" allowfullscreen frameborder="0"></iframe>
    <iframe src="https://clips.twitch.tv/embed?parent=ggs.sx&#x26;parent=example.com&#x26;clip=LovelyAstuteCoffeeImGlitch" width="560" height="315" allowfullscreen frameborder="0"></iframe>
    `)
  })

  test('set one twitch parent', () => {
    // Arrange
    let string = dedent`
    !(https://clips.twitch.tv/LovelyAstuteCoffeeImGlitch)
    `

    // Act
    let result = ggsmark(string, { twitchParents: ['ggs.sx'] })

    // Assert
    expect(result).toBe(dedent`
    <iframe src="https://clips.twitch.tv/embed?parent=ggs.sx&#x26;clip=LovelyAstuteCoffeeImGlitch" width="560" height="315" allowfullscreen frameborder="0"></iframe>
    `)
  })

  test('should show twitch video', () => {
    // Arrange
    let string = dedent`
    !(https://twitch.tv/videos/732347536)
    `

    // Act
    let result = ggsmark(string, { twitchParents: ['ggs.sx'] })

    // Assert
    expect(result).toBe(dedent`
    <iframe src="https://player.twitch.tv/?parent=ggs.sx&#x26;video=732347536" width="560" height="315" allowfullscreen frameborder="0"></iframe>
    `)
  })

  test('should show twitch channel', () => {
    // Arrange
    let string = dedent`
    !(https://twitch.tv/vinesauce)
    `

    // Act
    let result = ggsmark(string, { twitchParents: ['ggs.sx'] })

    // Assert
    expect(result).toBe(dedent`
    <iframe src="https://player.twitch.tv/?parent=ggs.sx&#x26;channel=vinesauce" width="560" height="315" allowfullscreen frameborder="0"></iframe>
    `)
  })

  test('should show twitch channel using player url', () => {
    // Arrange
    let string = dedent`
    !(https://player.twitch.tv/?channel=vinesauce)
    `

    // Act
    let result = ggsmark(string, { twitchParents: ['ggs.sx'] })

    // Assert
    expect(result).toBe(dedent`
    <iframe src="https://player.twitch.tv/?parent=ggs.sx&#x26;channel=vinesauce" width="560" height="315" allowfullscreen frameborder="0"></iframe>
    `)
  })

  test('should show twitch video using player url', () => {
    // Arrange
    let string = dedent`
    !(https://player.twitch.tv/?video=741520731)
    `

    // Act
    let result = ggsmark(string, { twitchParents: ['ggs.sx'] })

    // Assert
    expect(result).toBe(dedent`
    <iframe src="https://player.twitch.tv/?parent=ggs.sx&#x26;video=741520731" width="560" height="315" allowfullscreen frameborder="0"></iframe>
    `)
  })

  test('should show twitch video using player url', () => {
    // Arrange
    let string = dedent`
    !(https://player.twitch.tv/?video=741520731)
    `

    // Act
    let result = ggsmark(string, { twitchParents: ['ggs.sx'] })

    // Assert
    expect(result).toBe(dedent`
    <iframe src="https://player.twitch.tv/?parent=ggs.sx&#x26;video=741520731" width="560" height="315" allowfullscreen frameborder="0"></iframe>
    `)
  })
})

describe('render code blocks highlighted', () => {
  test('should highlight javascript', () => {
    // Arrange
    let string = dedent`
    \`\`\`javascript
    let dog = true

    if (dog) {
      console.log('woof')
    }
    \`\`\`
    `

    // Act
    let result = ggsmark(string, { twitchParents: ['ggs.sx'] })

    // Assert
    expect(result).toBe(dedent`
    <pre><code class="hljs language-javascript"><span class="hljs-keyword">let</span> dog = <span class="hljs-literal">true</span>

    <span class="hljs-keyword">if</span> (dog) {
      <span class="hljs-built_in">console</span>.log(<span class="hljs-string">'woof'</span>)
    }
    </code></pre>
    `)
  })
}),
describe('render medal.tv blocks', () => {
  test('single line', () => {
    // Arrange
    let string = dedent`
    !(https://medal.tv/clips/33631679/LrG6jg4Y2AUk)
    `

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toBe(dedent`
    <iframe src="https://medal.tv/clips/33631679/LrG6jg4Y2AUk" width="640" height="360" allowfullscreen frameborder="0"></iframe>
    `)
  })

  test('should not show iframe inline', () => {
    // Arrange
    let string =
      '**bold text** before text !(https://medal.tv/clips/33631679/LrG6jg4Y2AUk) after text **bold**'

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toBe(dedent`
    <p><strong>bold text</strong> before text !(<a href="https://medal.tv/clips/33631679/LrG6jg4Y2AUk">https://medal.tv/clips/33631679/LrG6jg4Y2AUk</a>) after text <strong>bold</strong></p>
    `)
  })

  test('multiple occurrences', () => {
    // Arrange
    let string = dedent`
    !(https://medal.tv/clips/33631679/LrG6jg4Y2AUk)
    !(https://medal.tv/clips/33631679/LrG6jg4Y2AUk)
    !(https://medal.tv/clips/33631679/LrG6jg4Y2AUk)
    `

    // Act
    let result = ggsmark(string)

    // Assert
    expect(result).toBe(dedent`
    <iframe src="https://medal.tv/clips/33631679/LrG6jg4Y2AUk" width="640" height="360" allowfullscreen frameborder="0"></iframe>
    <iframe src="https://medal.tv/clips/33631679/LrG6jg4Y2AUk" width="640" height="360" allowfullscreen frameborder="0"></iframe>
    <iframe src="https://medal.tv/clips/33631679/LrG6jg4Y2AUk" width="640" height="360" allowfullscreen frameborder="0"></iframe>
    `)
  })
})