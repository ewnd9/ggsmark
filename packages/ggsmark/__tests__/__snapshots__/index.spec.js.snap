// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render center alignment text multi line 1`] = `
"<div style=\\"text-align: center\\"><p>This is a multiple line texts
2nd line here</p></div>"
`;

exports[`render center alignment text single line 1`] = `"<div style=\\"text-align: center\\"><p>CenterText</p></div>"`;

exports[`render left alignment text multi line 1`] = `
"<div style=\\"text-align: left\\"><p>This is a multiple line texts
2nd line here</p></div>"
`;

exports[`render left alignment text single line 1`] = `"<div style=\\"text-align: left\\"><p>LeftText</p></div>"`;

exports[`render right alignment text multi line 1`] = `
"<div style=\\"text-align: right\\"><p>This is a multiple line texts
2nd line here</p></div>"
`;

exports[`render right alignment text single line 1`] = `"<div style=\\"text-align: right\\"><p>RightText</p></div>"`;

exports[`render soundcloud blocks multiple occurrences 1`] = `
"<iframe src=\\"https://w.soundcloud.com/player/?visual=true&#x26;url=https://soundcloud.com/iamcardib/wap-feat-megan-thee-stallion\\" width=\\"100%\\" height=\\"150\\" allowfullscreen frameborder=\\"0\\"></iframe>
<iframe src=\\"https://w.soundcloud.com/player/?visual=true&#x26;url=https://soundcloud.com/iamcardib/wap-feat-megan-thee-stallion\\" width=\\"100%\\" height=\\"150\\" allowfullscreen frameborder=\\"0\\"></iframe>
<iframe src=\\"https://w.soundcloud.com/player/?visual=true&#x26;url=https://soundcloud.com/iamcardib/wap-feat-megan-thee-stallion\\" width=\\"100%\\" height=\\"150\\" allowfullscreen frameborder=\\"0\\"></iframe>"
`;

exports[`render soundcloud blocks should not show iframe inline 1`] = `"<p><strong>bold text</strong> before text !(<a href=\\"https://soundcloud.com/iamcardib/wap-feat-megan-thee-stallion\\">https://soundcloud.com/iamcardib/wap-feat-megan-thee-stallion</a>) after text <strong>bold</strong></p>"`;

exports[`render soundcloud blocks single line 1`] = `"<iframe src=\\"https://w.soundcloud.com/player/?visual=true&#x26;url=https://soundcloud.com/iamcardib/wap-feat-megan-thee-stallion\\" width=\\"100%\\" height=\\"150\\" allowfullscreen frameborder=\\"0\\"></iframe>"`;

exports[`render youtube blocks repeated youtube 1`] = `
"<iframe src=\\"https://www.youtube.com/embed/52c_QSg64fs\\" width=\\"560\\" height=\\"315\\" allowfullscreen frameborder=\\"0\\"></iframe>
<iframe src=\\"https://www.youtube.com/embed/52c_QSg64fs\\" width=\\"560\\" height=\\"315\\" allowfullscreen frameborder=\\"0\\"></iframe>
<p>!(http://www.youshit.com)</p>"
`;

exports[`render youtube blocks repeated youtube with text before and after 1`] = `
"<p><strong>bold</strong> string before comyoutube !(<a href=\\"http://www.youtube.com/watch?v=52c_QSg64fs\\">http://www.youtube.com/watch?v=52c_QSg64fs</a>) after youtube !(<a href=\\"http://www.youtube.com/watch?v=waefawefwaef\\">http://www.youtube.com/watch?v=waefawefwaef</a>) <em>italics</em>
soft new line
new line</p>"
`;

exports[`render youtube blocks short-hand youtube 1`] = `"<p>!(http://www.youtu.be/52c_QSg64fs)</p>"`;

exports[`render youtube blocks youtube with timestamp 1`] = `"<iframe src=\\"https://youtube.com/embed/IJhgZBn-LHg?t=712\\" width=\\"560\\" height=\\"315\\" allowfullscreen frameborder=\\"0\\"></iframe>"`;
